File: String/validParanthesis.cpp
#include <bits/stdc++.h>
using namespace std;
bool isValid(string s)
{
    stack<char> st;
    for (int i = 0; i < s.length(); i++)
    {

        if (s[i] == '(' || s[i] == '{' || s[i] == '[')
        {
            st.push(s[i]);
        }

        else
        {
            if (st.empty())
                return false;
            if (s[i] == ')' && st.top() != '(')
                return false;
            if (s[i] == '}' && st.top() != '{')
                return false;
            if (s[i] == ']' && st.top() != '[')
                return false;
            st.pop();
        }
    }
    return st.empty();
}
int main()
{
    cout << isValid("{[]}");
    return 0;
}

File: String/validParanthesis.java
import java.util.Stack;

public class validParanthesis {
    public boolean isValid(String s) {
        Stack<Character> st = new Stack<Character>();
        for (char ch : s.toCharArray()) {
            if (ch == '(' || ch == '[' || ch == '{')
                st.push(ch);
            else {
                if (st.isEmpty())
                    return false;
                if (ch == ')' && st.peek() != '(')
                    return false;
                if (ch == ']' && st.peek() != '[')
                    return false;
                if (ch == '}' && st.peek() != '{')
                    return false;
                st.pop();
            }
        }
        return (st.isEmpty());
    }

    public boolean isValidP(String s) {
        while (true) {
            if (s.contains("()"))
                s = s.replace("()", "");
            else if (s.contains("[]"))
                s = s.replace("[]", "");
            else if (s.contains("{}"))
                s = s.replace("{}", "");
            else {
                return s.isEmpty();
            }
        }
    }
}
